server/py/services/api/tests/unit/test_initial_data.py

```diff
@@ -418,6 +418,82 @@ def test_init_system_id(
     assert system_id_after_second_init == system_id
 
 
+def test_ensure_latest_tag_for_artifacts():
```

Comment:
maybe you can add a summary (was a little confusing for me to follow at first):
```suggestion
Tests `_ensure_latest_tag_for_artifacts` to verify it reassigns the "latest" tag to the most 
recent artifact for each `(project, key, iteration)` combination when tags are missing.

Creates Artifacts:
- `(project1, key1, 0)`: Tags "v1", "v2".
- `(project1, key1, 1)`: Implicitly "latest".
- `(project1, key1, 2)`: Tag "v1".
- `(project2, key1, 0)`: Implicitly "latest".
- `(project2, key2, 0)`: Implicitly "latest".

Steps:
1. Verify initial state: Check 3 "latest"-tagged artifacts for `(project1, key1)`.
2. Remove "latest" from two artifacts and verify only one remains tagged.
3. Run migration with `chunk_size=1`.
4. Verify all 3 artifacts in `(project1, key1)` are correctly retagged as "latest".
```